src/Entity/PersistentEntity.php

```diff
@@ -202,7 +206,10 @@ private function castValues(array $attributes) {
 				$this->$attribute = (int)$this->$attribute;
 			} elseif ($definition[0] === T::Float) {
 				$this->$attribute = (float)$this->$attribute;
-			} else {
+			} elseif ($definition[0] === T::JSON) {
+				$serializer = new \Zumba\JsonSerializer\JsonSerializer();
+				$this->$attribute = $serializer->unserialize($this->$attribute);
```

Comment:
Are you very sure this doesn't open up a huge security hole? Serializing/deserializing arbitrary PHP object instances sounds pretty scary.